akka-remote/src/main/scala/akka/remote/artery/Handshake.scala

```diff
@@ -59,50 +59,74 @@ private[akka] class OutboundHandshake(outboundContext: OutboundContext, timeout:
       import OutboundHandshake._
 
       private var handshakeState: HandshakeState = Start
-
-      override def preStart(): Unit = {
-        val uniqueRemoteAddress = outboundContext.associationState.uniqueRemoteAddress
-        if (uniqueRemoteAddress.isCompleted) {
-          handshakeState = Completed
-        } else {
-          // The InboundHandshake stage will complete the uniqueRemoteAddress future
-          // when it receives the HandshakeRsp reply
-          implicit val ec = materializer.executionContext
-          uniqueRemoteAddress.foreach {
-            getAsyncCallback[UniqueAddress] { a ⇒
-              if (handshakeState != Completed) {
-                handshakeCompleted()
-                if (isAvailable(out))
-                  pull(in)
-              }
-            }.invoke
-          }
-        }
-      }
+      private var lastMessageTime = System.nanoTime()
+      private val injectHandshakeIntervalNanos = injectHandshakeInterval.toNanos
+      private var pendingMessage: Send = null
 
       // InHandler
       override def onPush(): Unit = {
         if (handshakeState != Completed)
           throw new IllegalStateException(s"onPush before handshake completed, was [$handshakeState]")
-        push(out, grab(in))
+
+        // inject a HandshakeReq once in a while to trigger a new handshake when destination
+        // system has been restarted
+        // FIXME if nanoTime for each message is too costly we could let the TaskRunner update
```

Comment:
Can't we just use a normal timer here? The injection interval is not too frequent so that should be fine I think.